src/control/cmd/dmg/pool.go

```diff
@@ -531,72 +531,104 @@ func (cmd *poolEvictCmd) Execute(args []string) error {
-	TargetIdx string `long:"target-idx" description:"Comma-separated list of target idx(s) to be excluded from the rank"`
+	TargetIdx string `long:"target-idx" description:"Comma-separated list of target index(es) to be excluded from each rank"`
 }
 
 // Execute is run when PoolExcludeCmd subcommand is activated
 func (cmd *poolExcludeCmd) Execute(args []string) error {
-	msg := "succeeded"
-
 	var idxList []uint32
 	if err := common.ParseNumberList(cmd.TargetIdx, &idxList); err != nil {
 		return errors.WithMessage(err, "parsing target list")
 	}
 
-	req := &control.PoolExcludeReq{ID: cmd.PoolID().String(), Rank: ranklist.Rank(cmd.Rank), TargetIdx: idxList, Force: cmd.Force}
+	req := &control.PoolRanksReq{
+		ID:        cmd.PoolID().String(),
+		Ranks:     cmd.RankList.Ranks(),
+		TargetIdx: idxList,
+		Force:     cmd.Force,
+	}
+
+	resp, err := control.PoolExclude(cmd.MustLogCtx(), cmd.ctlInvoker, req)
 
-	err := control.PoolExclude(cmd.MustLogCtx(), cmd.ctlInvoker, req)
+	if cmd.JSONOutputEnabled() {
+		return cmd.OutputJSON(resp, err)
+	}
+
+	cmd.Debugf("%T: %+v, %T: %+v", req, req, resp, resp)
+
+	// Retrieve PoolRanksResults so we can pretty print output.
+	results, err := resp.GetResults(err)
 	if err != nil {
-		msg = errors.WithMessage(err, "failed").Error()
+		cmd.Errorf(errors.WithMessage(err, "Pool exclude failed").Error())
+		return err
 	}
 
-	cmd.Infof("Exclude command %s\n", msg)
+	cmd.Debugf("%T: %+v", results, results)
 
-	return err
+	var out strings.Builder
+	pretty.PrintPoolRankResults(&out, results)
+	cmd.Info(out.String())
+
+	return resp.Errors()
 }
 
 // poolDrainCmd is the struct representing the command to Drain a DAOS target.
 type poolDrainCmd struct {
-	poolCmd
-	Rank      uint32 `long:"rank" required:"1" description:"Engine rank of the targets to be drained"`
-	TargetIdx string `long:"target-idx" description:"Comma-separated list of target idx(s) to be drained on the rank"`
+	poolRanksCmd
+	TargetIdx string `long:"target-idx" description:"Comma-separated list of target index(es) to be drained on each rank"`
 }
 
 // Execute is run when PoolDrainCmd subcommand is activated
 func (cmd *poolDrainCmd) Execute(args []string) error {
-	msg := "succeeded"
-
 	var idxList []uint32
 	if err := common.ParseNumberList(cmd.TargetIdx, &idxList); err != nil {
-		err = errors.WithMessage(err, "parsing target list")
-		return err
+		return errors.WithMessage(err, "parsing target list")
 	}
 
-	req := &control.PoolDrainReq{
+	req := &control.PoolRanksReq{
 		ID:        cmd.PoolID().String(),
-		Rank:      ranklist.Rank(cmd.Rank),
+		Ranks:     cmd.RankList.Ranks(),
 		TargetIdx: idxList,
 	}
 
-	err := control.PoolDrain(cmd.MustLogCtx(), cmd.ctlInvoker, req)
+	resp, err := control.PoolDrain(cmd.MustLogCtx(), cmd.ctlInvoker, req)
+
+	if cmd.JSONOutputEnabled() {
+		return cmd.OutputJSON(resp, err)
+	}
+
+	cmd.Debugf("%T: %+v, %T: %+v", req, req, resp, resp)
```

Comment:
Are these debug lines necessary? I think you should already have debug printing from the control library.

Reply:
I wasn't getting any debug for response from `interceptors.go` but  maybe I can just move this to trace. this also prints the native types rather than proto which is what we get from the control library from what I can see